OBJLoader2 V2.3.1: Parser Verification, Point and Line support, moved generic functions to LoaderBase

kaisalmen opened the pull request:
These are the enhancements and bug fixes:
- Org Repo Issue 28: Parser Validation. Parser now is one single class. Simplified code, fixed bugs missing 'g' statement, etc. and introduced obj_obj2 verification examples with gulpfile to re-create verify.obj.
- Org Repo Issue 31: Worker code still works when mangling is used during minification. Needed to remove Consts.
- Org Repo Issue 32: THREE.LoaderSupport.ConsoleLogger: Allow to pass additional arguments to error, warn, info and debug
- Org Repo Issue 33: Added support for Points and fixed Lines along
- Fixed CRLF issues. Now all files has unix line-endings.

Hi @mrdoob, here comes an important update. I have finished a Parser review. It is now one single object. The code should be easier to understand for others, I think.
I have introduced `verify.obj` which can be rebuilt with gulp instructions. It is is used by new example `obj_obj2_verify`. This should be used and expanded in the future to highlight & fix Parser issues. `OBJLoader` has some (Omission of 'g' and usage of different face descriptions). `OBJLoader2` is now able to handle those cases.

kaisalmen commented:
Do you have any idea why the checks were unable to run?

kaisalmen commented:
I amended the commit and push it again. Putting the Parser into its own namespace is no longer required (work-around no longer required to overcome minification problems).

mrdoob commented on examples/files.js:
```diff
@@ -107,6 +107,7 @@ var files = {
 		"webgl_loader_nodes",
 		"webgl_loader_obj",
 		"webgl_loader_obj_mtl",
+		"webgl_loader_obj_obj2_verify",
```
How about `webgl_loader_obj2_verify`?

kaisalmen replied:
It uses both `OBJLoader` and `OBJLoader2` that's why I chose that name. You decide!

mrdoob commented on examples/webgl_loader_obj2_meshspray.html:
```diff
@@ -1,521 +1,517 @@
+
+						return workerCode;
+					};
+					var libs2Load = [ 'node_modules/three/build/three.js' ];
```
`node_modules/three/build/three.js`?

kaisalmen replied:
It should be `build/three.min.js`. Thanks.
There are other path issues in other examples. I will double check now and push an update.

kaisalmen commented:
I fixed the path issues, but the verify.obj is not loaded properly. Did someone break `MTLLoader`. When I don't assign materials the cubes are displayed.

kaisalmen commented:
Some background info on the mtl problem. I manually created the mtl file and I think I have started of one that was created by 3dsmax. `Tr` was set to 0. There is no need for transparency in my current test case (not yet), but after now checking #12862 I wonder why the behaviour changes if you have not set the option in `MTLLoader`. This smells....

kaisalmen commented:
@mrdoob I pushed another update that adds Point support `p` support and it fixes Line creation.
Line and Point Support are available in the global `THREE.LoaderSupport.Builder`. User just needs to ensure proper Material are used, This is done by `OBJLoader2` and also more globally available if a Loader is based on `THREE.LoaderSupport.LoaderBase`.

From my point of view this is ready. 👍 
Do you know why the code checks fail? This seems to be not code related.

kaisalmen commented:
Ok, here are some more improvements resulting from work on worker based PCDLoader.

kaisalmen commented:
V2.3.0 is now available via npm:
https://www.npmjs.com/package/wwobjloader2

kaisalmen commented:
In preparation of PCDLoader PR (add parseAsync and run) which will come tomorrow (need to update docs as well and ran out of time) I have pushed another update bumping the minor version up.
Basically, generic functions `load` and `checkResourceDescriptorFiles` extracted from `OBJLoader2`  are now available in `THREE.LoaderSupport.LoaderBase`.

@mrdoob do you see this whole PR merged before R90? Thanks

kaisalmen committed:
#13156 OBJLoader2 V2.3.1:
OrgRepoIssue 28: Parser Validation. Parser now is one single class: Simplified code, fixed bugs missing 'g' statements and introduced obj_obj2 verification with gulpfile to re-create verify.obj.
OrgRepoIssue 31: Worker code still works when mangling is used during minification. Needed to remove Consts.
OrgRepoIssue 32: THREE.LoaderSupport.ConsoleLogger: Allow to pass additional arguments to error, warn, info and debug
OrgRepoIssue 33: Added support for Points and fixed Lines along
- Removed Tr instruction from verify.mtl
- WorkerSupport.buildSingelton allows to use constructors from functions list (fixed singleton typo along)
- Builder defines the default materials and renamed 'vertexColorMaterial' to 'defaultVertexColorMaterial' for consistency.
- Fixed Builder.updateMaterials clone instructions
- Extracted load and checkResourceDescriptorFiles to LoaderBase.
- loadMtl now allows to pass material options to MTLLoader
- Updated documentation

kaisalmen commented:
@mrdoob I squashed down all commits to one and put it at the head of dev (sorry, needed to force push, again as I was behind).
I moved `verify.html` alongside the `verify.obj` to `examples/obj/verify` and removed it from the example. This really is more a test than an example.

mrdoob commented:
> I moved `verify.html` alongside the `verify.obj` to `examples/obj/verify` and removed it from the example. 

Do you mind moving these to `examples/models/obj/verify` instead? Ideally we'll kill that `examples/obj` folder soon.

kaisalmen committed:
#1356 Moved verify and separated point from line cube

kaisalmen commented:
Done, separated point and line cubes in verify.obj in addition.

mrdoob commented:
Sweet!

mrdoob commented:
Thanks!
